### PR TITLE
upgrade to h3 v4.2.0

### PR DESCRIPTION
**Pull request by isaacbrodsky**

### CODE REVIEW

File: src/h3/api/basic_int/__init__.py [hunk 1]
```diff
@@ -477,6 +477,10 @@ def h3shape_to_cells(h3shape, res):
     h3shape : ``H3Shape``
     res : int
         Resolution of the output cells
+    flags : int
+        Containment mode flags
```
Comment (isaacbrodsky) [inline src/h3/api/basic_int/__init__.py:481]: `flags` seems like it needs an enum

**Comment by isaacbrodsky** [comment]
Coverage seems to be getting 403 which I don't think is related to this change.

File: tests/test_cells_and_edges.py [hunk 2]
```diff
@@ -289,11 +289,11 @@ def test_average_hexagon_area():
...
-        0: 1107.712591000,
-        1: 418.676005500,
-        2: 158.244655800,
-        9: 0.174375668,
-        15: 0.000509713,
+        0: 1281.256011,
+        1: 483.0568391,
+        2: 182.5129565,
+        9: 0.200786148,
+        15: 0.000584169,
```
Comment (ajfriend) [inline tests/test_cells_and_edges.py:296]: This is a pretty big change. Do we know what happened here?

Comment (isaacbrodsky) [reply tests/test_cells_and_edges.py:296]: This appears to be reflecting https://github.com/uber/h3/pull/749

Comment (ajfriend) [reply tests/test_cells_and_edges.py:296]: OK, I guess I thought we had this in a while ago.

**Comment by ajfriend** [comment]
I was looking for docs on the containment modes and realized that we don't have it documented. As part of this effort, I think we should do:

- https://github.com/uber/h3/issues/956
- https://github.com/uber/h3/issues/957

File: src/h3/api/basic_int/__init__.py [hunk 3]
```diff
@@ -467,7 +467,7 @@ def uncompact_cells(cells, res):
     return _out_collection(hu)
 
 
-def h3shape_to_cells(h3shape, res):
+def h3shape_to_cells(h3shape, res, flags=0, experimental=False):
```
Comment (ajfriend) [inline src/h3/api/basic_int/__init__.py:470]: I'm not sure if this should be the final form of the API. For example, we might not use integers for the flags, and we may drop the `experimental` option.

Could we leave `h3shape_to_cells` unchanged, and just create a `h3shape_to_cells_experimental` where we're free to change the API?

**Review by dfellis — APPROVED** [review]

**Comment by ajfriend** [comment]
This looks great. thanks!

**Pull request merged**